### DISCUSSION TITLE
Nextjs ant design

### DISCUSSION THREAD

**Discussion by sadam1807** [vercel/next.js#15530] [Help]
I clone the repo from next js examples to integrate nextjs with antd design. 

Every thing works fine but css is loading after html send to the browser. 

How I can resolve this issue

**Comment by modbender** [comment]
Check this out once: https://medium.com/@tanakorn0412/customize-ant-design-theme-in-next-js-version-9-5958fbc531a3

**Comment by sadam1807** [comment]
@modbender thanks for your response. but still I am facing the issue. CSS is flickering, it takes time to load because HTML is sent before the CSS to browser, the issue is not only with antd CSS  but also with all the parent css files loaded in _app.js file.

**Comment by modbender** [comment]
Oh ok, then guess you'll have to ping/tag some other devs or contributors to get a reply...

**Comment by sadam1807** [comment]
@timneutkens @viztor @mvfsillva @awareness481 @jacklee814 @mit123suki

**Comment by awareness481** [comment]
@sadam1807 
This was fixed in the **Next.js 9.5** version, either update your *package.json* to 
`"next": "latest"` and re-install your node-modules     
**or** create a new Next.js app using `npx create-next-app` which should use the latest version.

Please refrain from tagging too many people tho.

**Reply by sadam1807** [reply]
@awareness481  thanks for your response  and sorry for tagging many people
i already have the latest version install, I cloned the https://github.com/vercel/next.js/tree/canary/examples/with-ant-design repo but still, I am facing the issue.
![image](https://user-images.githubusercontent.com/42937204/89179558-955a4100-d5ad-11ea-9e86-dc33470a662d.png)

https://nextjs-test-livid.vercel.app/


when you load the website you can observe that css is flickering

**Reply by awareness481** [reply]
Oh I see it now. This isn't the issue with Next.js that I mentioned before, it's the Antd animations.

If you want to see which animations are running, you can click on the "3 dots" (on the right) on the chrome console -> More tools -> Animations.

![chrome dev tools, animation](https://user-images.githubusercontent.com/12380586/89181970-64721000-d59d-11ea-8ce6-51d7de014e92.png)

There you can inspect animations and disable the ones you want. A quick, hacky way to disable an animation is to you use 
`animation-duration: 0s !important;`

For example if you use 
```css
* {
  animation-duration: 0s !important;
}
```
it will disable all animations (the * matches all elements). Obviously you might not want that but give it a try and see if it solves your issue.
I'm not sure if the Antd API provides a way to disable transitions.

Tell me if this answers your question.

**Reply by sadam1807** [reply]
@awareness481 added the code provided by you in vars.css and re-ordered the  import in _app.js.
import '../styles/vars.css'
import '../styles/global.css'
import 'antd/dist/antd.css'

export default function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

but still the same result.

**Comment by mit123suki** [comment]
@sadam1807  Looks like styles are not loaded in proper order. 

Anyway, try this setup https://github.com/mit123suki/next-ant-setup, it solves this flickering problems and also allows you to customize theme without touching webpack config

**Comment by sadam1807** [comment]
@mit123suki thanks for your response but I am getting an error with you code.

![image](https://user-images.githubusercontent.com/42937204/89276794-afedf200-d661-11ea-8efd-2d5641b781ef.png)

**Reply by mit123suki** [reply]
@sadam1807 It's not related to my setup. It's something wrong in your codebase, error clearly says invalid hook call. Your SS doesn't help at all. BTW let's not extend this discussion here as this is totally irrelevant to the issue here

**Comment by sadam1807** [comment]
@mit123suki  I fixed that  but how i can change the color of antd component, I try changing the code from files mytheme.less and sass-vars.scss but its not working.
and also the date picker css is breaking.

![image](https://user-images.githubusercontent.com/42937204/89282008-85ebfe00-d668-11ea-946a-ff06b7d59ae2.png)

**Comment by mit123suki** [comment]
@sadam1807  You seem to not follow any instructions detailed in readme file, please do that and come back if you have any issues. You don't modify sass-vars.scss, it's generated by the script, and you reuse it. Only file you touch is mytheme.less, where you make changes and are reflected in real time only if you follow the instructions and do setup accordingly

Regarding the datepicker issue, it's related to ant dayjs plugin if i remember correctly, do check the website and follow the fix

Rather than working on your repo, clone the given repo and play with it and understand how it is setup and works, this way you know whether the issues you get is related to setup or next or ant and to know where to post and get the solution exactly.

**Comment by sadam1807** [comment]
@mit123suki from the beginning I was using your repo, I fixed hooks issue changing ant-d version.
If I followed you read me instruction , will create two vscode and less-script files.
